pypesto/petab/importer.py

```diff
@@ -78,6 +81,74 @@ def from_yaml(yaml_config: Union[dict, str],
             output_folder=output_folder,
             model_name=model_name)
 
+    def check_gradients(
+        self,
+        *args,
+        rtol: float = 1e-2,
+        atol: float = 1e-3,
+        mode: Literal = None,
+        objective: Objective = None,
+        objAbsoluteTolerance: float = 1e-10,
+        objRelativeTolerance: float = 1e-12,
+        multi_eps=None,
+        **kwargs
+    ):
+        """Check if gradients match finite differences (FDs)
+        Parameters
+        ----------
+        rtol: relative error tolerance
+        atol: absolute error tolerance
+        mode: unction values or residuals
+        objective: objective function
+        objAbsoluteTolerance: absolute tolerance in sensitivity calculation
+        objRelativeTolerance: relative tolerance in sensitivity calculation
+        multi_eps: multiple test step width for FDs
+
+        Returns
+        -------
+        bool
+            Indicates whether gradients match (True) FDs or not (False)
+        """
+        par = np.asarray(self.petab_problem.x_nominal_scaled)
+        problem = self.create_problem()
+
+        if objective is None:
+            objective = problem.objective
+            objective.amici_solver.setSensitivityMethod(
+                amici.SensitivityMethod_forward)
+            objective.amici_solver.setAbsoluteTolerance(objAbsoluteTolerance)
+            objective.amici_solver.setRelativeTolerance(objRelativeTolerance)
+
+        free_indices = par[problem.x_free_indices]
+        dfs = []
+        modes = []
+
+        if mode is None:
+            modes = [MODE_FUN, MODE_RES]
+        else:
+            modes = [mode]
+
+        if multi_eps is None:
+            multi_eps = [1e-3, 1e-4, 1e-5]
+
+        for mode in modes:
+            try:
+                dfs.append(objective.check_grad_multi_eps(
+                            free_indices, *args, **kwargs,
+                            mode=mode, multi_eps=multi_eps))
+            except (RuntimeError, ValueError):
+                # Might happen in case PEtab problem not well defined or
+                # fails for specified tolerances in forward sensitivities
+                pass
+
+        return any([
```

Comment:
```suggestion
        return all([
```

Shouldn't that be `all`? If it fails for one mode, I'd consider that a failure.

Reply:
Yes it should be all, you are right.